src/main/java/org/plumelib/bcelutil/StackMapUtils.java

```diff
@@ -0,0 +1,1205 @@
+        }
+      } else {
+        // If there were compiler temps prior to this local, we don't
+        // need to worry about them as the compiler will have already
+        // included them in the StackMaps. Reset the indicators.
+        compiler_temp_i = -1;
+      }
+    }
+
+    // We have looked at all the local variables; if we have not found
+    // a place for our new local, check to see if last local was a
+    // compiler temp and insert prior.
+    if ((new_offset == -1) && (compiler_temp_i != -1)) {
+      new_offset = locals[compiler_temp_i].getIndex();
+      new_index = compiler_temp_i;
+    }
+
+    // If new_offset is still unset, we can just add our local at the
+    // end.  There may be unnamed compiler temps here, so we need to
+    // check for this (via max_offset) and move them up.
+    if (new_offset == -1) {
+      new_offset = max_offset;
+      if (new_offset < mgen.getMaxLocals()) {
+        mgen.setMaxLocals(mgen.getMaxLocals() + local_type.getSize());
+      }
+      lv_new = mgen.addLocalVariable(local_name, local_type, new_offset, null, null);
+    } else {
+      // insert our new local variable into existing table at 'new_offset'
+      lv_new = mgen.addLocalVariable(local_name, local_type, new_offset, null, null);
+      // we need to adjust the offset of any locals after our insertion
+      for (i = new_index; i < locals.length; i++) {
+        LocalVariableGen lv = locals[i];
+        lv.setIndex(lv.getIndex() + local_type.getSize());
+      }
+      mgen.setMaxLocals(mgen.getMaxLocals() + local_type.getSize());
+    }
+
+    debug_instrument.log(
+        "Added local  %s%n", lv_new.getIndex() + ": " + lv_new.getName() + ", " + lv_new.getType());
+
+    // Now process the instruction list, adding one to the offset
+    // within each LocalVariableInstruction that references a
+    // local that is 'higher' in the local map than new local
+    // we just inserted.
+    adjust_code_for_locals_change(mgen, new_offset, local_type.getSize());
+
+    // Finally, we need to update any FULL_FRAME StackMap entries to
+    // add in the new local variable type.
+    update_full_frame_stack_map_entries(new_offset, local_type, locals);
+
+    debug_instrument.log("New LocalVariableTable:%n%s%n", mgen.getLocalVariableTable(pool));
+    return lv_new;
+  }
+
+  /**
+   * Under some circumstances, there may be problems with the local variable table.
+   *
+   * <ol>
+   *   <li>In some special cases where parameters are added by the Java compiler (eg, constructors
+   *       for inner classes), the local variable table is missing the entry for this additional
+   *       parameter.
+   *   <li>The Java compiler allocates unnamed local temps for:
+   *       <ul>
+   *         <li>saving the exception in a finally clause
+   *         <li>the lock for a synchronized block
+   *         <li>interators
+   *         <li>(others?)
+   *       </ul>
+   *       We will create a 'fake' local for these cases.
```

Comment:
This comment refers only to the second element of the ordered list.  Also document what the method does for the first element ("In some special cases...").